### PR TITLE
Remove deprecated and unused definitions

### PR DESCRIPTION
**Pull request by lehins**
# Description

This PR removes all of the deprecated definitions and some other unused ones from:

* `cardano-ledge-core`
* `cardano-ledger-binary`
* `cardano-ledger-shelley`
* `cardano-ledger-conway`

The reason why other packages didn't get their deprecated definitions removed was because currently they don't have any other breaking changes in them, so such removal is postponed until later.

The easiest way to review, probably, is on per commit basis.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages.
      **_New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated.
      **_If you change the bounds in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

**Review by aniketd — APPROVED** [review]
💪 😓  LGTM!

File: libs/cardano-ledger-api/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Version history for `cardano-ledger-api`
 
-## 1.9.4.1
+## 1.10.0.0
```
Comment (teodanciu) [inline libs/cardano-ledger-api/CHANGELOG.md:3]: We should change the version in the cabal file too, no?

**Review by teodanciu — COMMENTED** [review]
Sorry, I didn't get to go through all commits yet, but I don't want to block the merging! 
I will look later, and in the unlikely event I'll find a problem, we can fix it later. 

The only thing so far I think we missed a version bump in the cabal file in cardano-ledger-api.

File: libs/cardano-ledger-api/CHANGELOG.md (hunk 1 above)
Comment (lehins) [reply libs/cardano-ledger-api/CHANGELOG.md:3]: We most certainly should. Thank you. Fixed.

File: eras/shelley/impl/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,26 @@
 
 ## 1.15.0.0
 
+* Remove deprecated `witsVKeyNeededGov`, `witsVKeyNeededNoGov`, `shelleyWitsVKeyNeeded` and `propWits`
```
Comment (teodanciu) [inline eras/shelley/impl/CHANGELOG.md:5]:
```suggestion
* Change parameter type  of `StakeKeyNonZeroAccountBalanceDELEG from  `Maybe Coin` to `Coin`  
* Remove deprecated `witsVKeyNeededGov`, `witsVKeyNeededNoGov`, `shelleyWitsVKeyNeeded` and `propWits`
```

**Review by teodanciu — APPROVED** [review]
Looks good to me - made just a little suggestion about the CHANGELOG.

**Pull request merged**